### PR TITLE
added condition to check offer only messages and messages send to don…

### PR DESCRIPTION
**Pull request by namrataukirde**
Hi Team,

Additional changes to Send SMS reminder considering 
- No private messages.
- Offers only messages.
-  Donor having other roles will be notified for offers created by him. (i.e donor only offers.)

Please review.

Thanks.

### CODE REVIEW

File: app/services/subscriptions_reminder.rb [hunk 1]
```diff
@@ -9,25 +9,32 @@ def generate
...
     states = ['submitted', 'under_review', 'reviewed', 'scheduled', 'received',
       'receiving', 'inactive'] # NOT draft, closed or cancelled
     user_ids = Offer.where(state: states).distinct.pluck(:created_by_id)
-    User.joins(subscriptions: [:message]).
+    User.joins(subscriptions: [:message, :offer]).
       where('users.id IN (?)', user_ids).
       where("COALESCE(users.sms_reminder_sent_at, users.created_at) < (?)", delta.iso8601).
       where('subscriptions.state': 'unread').
       where("messages.created_at > COALESCE(users.sms_reminder_sent_at, users.created_at)").
+      where("messages.offer_id IS NOT NULL OR messages.item_id IS NOT NULL and messages.order_id IS NULL").
```
Comment (namrataukirde) [inline app/services/subscriptions_reminder.rb]: only `order_id` null can work but added extra checks and I don't see any harm in it. :)

**Comment by namrataukirde** [comment]
I am modifying the same class to send slack notifications for staging testing purpose and committing to same PR. It will be staging only.

**Comment by namrataukirde** [comment]
Yup Sure 👍

File: spec/services/subscriptions_reminder_spec.rb [hunk 2]
```diff
@@ -127,6 +137,18 @@
...
+      it 'donor has order unread messages' do 
+        charity.update_column(:sms_reminder_sent_at, before_delta.ago)
+        expect(subject.send(:user_candidates_for_reminder).to_a).to eql([])
+      end
+
+      it 'charity user who is also donor has no unread messages on the offer but on order' do
+        msg4 = create(:message, offer: charity_offer, sender: reviewer)
+        charity.update_column(:sms_reminder_sent_at, before_delta.ago) 
+        charity.offers.first.subscriptions.unread.first.update_column(:state, 'read')
+        expect(subject.send(:user_candidates_for_reminder).to_a).to eql([])
```
Comment (abhinav098) [inline spec/services/subscriptions_reminder_spec.rb]: Good coverage @namrataukirde 👍

File: app/services/subscriptions_reminder.rb [hunk 3]
```diff
@@ -28,7 +28,7 @@ def user_candidates_for_reminder
         .where("COALESCE(users.sms_reminder_sent_at, users.created_at) < (?)", delta.iso8601)
         .where('subscriptions.state': 'unread')
         .where("messages.created_at > COALESCE(users.sms_reminder_sent_at, users.created_at)")
-        .where("messages.offer_id IS NOT NULL OR messages.item_id IS NOT NULL and messages.order_id IS NULL")
+        .where("(messages.offer_id IS NOT NULL OR messages.item_id IS NOT NULL) and messages.order_id IS NULL")
```
Comment (steveyken) [inline app/services/subscriptions_reminder.rb]: :+1: for clarity!

**Pull request merged**